frontend/src/scenes/sessions/SessionDetails.tsx

```diff
@@ -16,6 +19,7 @@ export function SessionDetails({ session }: { session: SessionType }): JSX.Eleme
     const [page, setPage] = useState(1)
     const [pageSize, setPageSize] = useState(50)
     const events = session.events || filteredSessionEvents[session.global_session_id]
+    const matchingEventIds = new Set(session.matching_events || [])
```

Comment:
Note: In the ideal world this belongs in the logic, not the component. Every re-render it would recalculate this. In practice this should be quite cheap though unless we rerender in a loop.

Reply:
Suggestion: `useMemo`?